custom_components/frigidaire/humidifier.py

```diff
@@ -165,36 +166,43 @@ def mode(self):
 
     @property
     def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Add extra state attributes specific to Frigidaire dehumidifiers"""
         fan_speed = self._details.for_code(
             frigidaire.HaclCode.AC_FAN_SPEED_SETTING
         ).number_value
 
-        return {
+        attrib = {
             "current_humidity": self._details.for_code(
                 frigidaire.HaclCode.AMBIENT_HUMIDITY
             ).number_value,
             "check_filter": bool(
                 self._details.for_code(
                     frigidaire.HaclCode.AC_CLEAN_FILTER_ALERT
                 ).number_value
-            ),   
-            "bin_full": bool(
+            ),
+            "fan_mode": FRIGIDAIRE_TO_HA_FAN_MODE[fan_speed],
+        }
+
+        # The following attributes only exist on some models of dehumidifier
+        if self._details.for_code(frigidaire.HaclCode.BIN_FULL_ALERT) is not None:
```

Comment:
Can you make these a bit more dry by assigning to a local variable instead of running for_code twice for each?

Reply:
I can, if that is what you prefer. What version of python are you targeting?

Reply:
I think HA is on 3.10 so that version should be good to target

Reply:
changes made